src/object_store/src/object/mod.rs

```diff
@@ -418,54 +430,49 @@ impl MonitoredStreamingReader {
         object_store_metrics: Arc<ObjectStoreMetrics>,
         streaming_read_timeout: Option<Duration>,
     ) -> Self {
-        let operation_type = "streaming_read";
-        let timer = object_store_metrics
-            .operation_latency
-            .with_label_values(&[media_type, operation_type])
-            .start_timer();
         Self {
             inner: handle,
             object_store_metrics,
             operation_size: 0,
             media_type,
-            timer: Some(timer),
             streaming_read_timeout,
+            operation_type_str: OperationType::StreamingRead.as_str(),
         }
     }
 
     pub async fn read_bytes(&mut self) -> Option<ObjectResult<Bytes>> {
-        let operation_type = "streaming_read_read_bytes";
         let _timer = self
             .object_store_metrics
             .operation_latency
-            .with_label_values(&[self.media_type, operation_type])
+            .with_label_values(&[self.media_type, self.operation_type_str])
             .start_timer();
         let future = async {
             self.inner
                 .next()
-                .verbose_instrument_await("object_store_streaming_read_read_bytes")
+                .verbose_instrument_await(self.operation_type_str)
                 .await
         };
         let res = match self.streaming_read_timeout.as_ref() {
             None => future.await,
             Some(timeout) => tokio::time::timeout(*timeout, future)
                 .await
                 .unwrap_or_else(|_| {
-                    Some(Err(ObjectError::internal(
-                        "streaming_read read_bytes timeout",
-                    )))
+                    Some(Err(ObjectError::timeout(format!(
+                        "{} timeout",
+                        self.operation_type_str
+                    ))))
                 }),
         };
```

Comment:
https://github.com/risingwavelabs/risingwave/pull/16231#discussion_r1570656358. I think we can deprecate this and ensure all streaming reader related timeout is maintained in the specific implementaion.